From fe8b6eff4fe5db6f598769970094251fb4a9e54e Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Tue, 18 Jan 2022 17:26:35 +0200
Subject: [PATCH] arm: mx6: cm-fx6: tag u-boot version -cm-fx6-4.1-phy

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e4eaaf8c5f..309879e241 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 2015
 PATCHLEVEL = 07
 SUBLEVEL =
-EXTRAVERSION = -cm-fx6-3.1
+EXTRAVERSION = -cm-fx6-4.1-phy
 NAME =
 
 # *DOCUMENTATION*
-- 
2.29.0

